src/gwt/src/org/rstudio/core/client/virtualscroller/virtualscroller.js

```diff
@@ -0,0 +1,299 @@
+/*
+ * virtualscroller.js
+ *
+ * Copyright (C) 2020 by RStudio, PBC
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+var VirtualScroller;
+
+(function () {
+  "use strict";
+  VirtualScroller = function () {
+  };
+
+  VirtualScroller.prototype = {
+    setup: function(element) {
+      if (!element)
+        return;
+
+      //  *** _CONSTANTS ***
+      this._DEBUG = true;
+      this._SCROLL_DEBOUNCE_MS = 500;
+      this._BUCKET_MAX_SIZE = 50;
+      this._MAX_VISIBLE_BUCKETS = 10;
+
+      // we use this style to keep the elements in the DOM for screen readers
+      this._HIDDEN_STYLE = "org-rstudio-core-client-theme-res-ThemeStyles-visuallyHidden";
```

Comment:
Have you tested this in a production build? GWT obfuscates class names, so you will either need to pass the class name into the setup method at runtime or tell GWT to use a non-obfuscated name.